src/commands/captureDiagnostics.ts

```diff
@@ -0,0 +1,209 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the VS Code Swift open source project
+//
+// Copyright (c) 2021-2024 the VS Code Swift project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of VS Code Swift project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import * as fs from "fs/promises";
+import * as path from "path";
+import * as vscode from "vscode";
+import { tmpdir } from "os";
+import { exec } from "child_process";
+import { Writable } from "stream";
+import { SwiftOutputChannel } from "../ui/SwiftOutputChannel";
+import { WorkspaceContext } from "../WorkspaceContext";
+import { Version } from "../utilities/version";
+import { execFileStreamOutput } from "../utilities/utilities";
+import configuration from "../configuration";
+
+export async function captureDiagnostics(ctx: WorkspaceContext) {
+    const diagnosticsDir = path.join(
+        tmpdir(),
+        `vscode-diagnostics-${formatDateString(new Date())}`
+    );
+
+    try {
+        await fs.mkdir(diagnosticsDir);
+        await writeLogFile(diagnosticsDir, "logs.txt", extensionLogs(ctx));
+        await writeLogFile(diagnosticsDir, "environment.txt", environmentLogs(ctx));
+
+        if (ctx.swiftVersion.isGreaterThanOrEqual(new Version(6, 0, 0))) {
+            // sourcekit-lsp diagnose command is only available in 6.0 and higher.
+            // await writeLogFile(diagnosticsDir, "sourcekit-lsp.txt", );
+            await sourcekitDiagnose(ctx, diagnosticsDir);
+        } else {
+            await writeLogFile(diagnosticsDir, "sourcekit-lsp.txt", sourceKitLogs(ctx));
+        }
+
+        await writeLogFile(diagnosticsDir, "diagnostics.txt", diagnosticLogs());
```

Comment:
Yes, I agree this is redundant if a `sourcekit-lsp diagnose` bundle is included, but if it isn't available it could maybe be useful.

Reply:
OK, I think we’re talking about different things here. What does `diagnostics.txt` include?

Reply:
Its essentially the contents of the Problems panel

Reply:
OK, then it shouldn’t really be redundant if `sourcekit-lsp diagnose` is included because it will also include eg. build issues. That’s what threw me off for a short bit. 

That’s why I was suggesting to rename it to `source-code-diagnostics.txt` to avoid any ambiguities with the diagnose command added by this PR and the `sourcekit-lsp diagnose` bundle.